pkg/controller/database/cockroach/cockroach_test.go

```diff
@@ -70,7 +85,7 @@ func cockroachCluster(im ...cockroachClusterModifier) *v1alpha1.CockroachCluster
 		Spec: v1alpha1.CockroachClusterSpec{
 			ResourceSpec: runtimev1alpha1.ResourceSpec{
 				ProviderReference:                &corev1.ObjectReference{Namespace: namespace, Name: providerName},
-				WriteConnectionSecretToReference: corev1.LocalObjectReference{Name: connectionSecretName},
+				WriteConnectionSecretToReference: &runtimev1alpha1.SecretReference{Name: connectionSecretName},
```

Comment:
Probably not strictly necessary, but I've been updating all these `SecretReference` test structs to include the secret's `Namespace` as well as its `Name`.